guides/plugins/plugins/framework/event/finding-events.md

```diff
@@ -0,0 +1,184 @@
+
+<container xmlns="http://symfony.com/schema/dic/services"
+           xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+           xsi:schemaLocation="http://symfony.com/schema/dic/services http://symfony.com/schema/dic/services/services-1.0.xsd">
+
+    <services>
+        <service id="Some\Service">
+            <argument type="service" id="Another/Service"/>
+            <argument type="service" id="event_dispatcher"/>
+        </service>
+    </services>
+</container>
+```
+
+Therefore, you could simply search for occurrences of the `event_dispatcher` in the respective `.xml` files.
+
+You can also do this the other way around, by having a look at the service's constructor parameters.
+
+```php
+public function __construct(
+    Some\Service $someService,
+    EventDispatcherInterface $eventDispatcher
+) {
+    $this->someService = $someService;
+    $this->eventDispatcher = $eventDispatcher;
+}
+```
+
+If it's having access to the `EventDispatcherInterface`, you're most likely going to find at least one event being fired
+in that service.
+
+## Storefront events
+
+We're also making use of events in our Storefront javascript plugins.
+We've already covered Storefront events in this [guide](../../storefront/reacting-to-javascript-events.md).
+
+However, it's not really explaining how you can find them in the first place.
+For this case, we're using the same plain method like before: Simply searching for them or by looking through the code.
+
+### Finding events in the code
+
+In the Storefront javascript plugins, you can notice custom events by the following pattern: 
+```javascript
+this.$emitter.publish('someEvent', additionalData);
+```
+
+Therefore, you could subscribe to the event named `someEvent` and gain access to `additionalData` here.
+
+```javascript
+this.$emitter.subscribe('someEvent', (additionalData) => {
+    // Do stuff
+});
+```
+
+### Searching for javascript events
+
+Searching for the said javascript events is done by searching for the following term in either the `<shopware root>/platform/src/Administration/Resources/app/administration` directory for
+the [development template](https://github.com/shopware/development) or the `<shopware root>/vendor/shopware/platform/src/Administration/Resources/app/administration` directory
+for the [production template](https://github.com/shopware/production):
+`$emitter.publish`.
+This way, you'll find all occurrences of plugins actually firing a custom event.
+
+## Administration events
+
+In the Administration, most events you can find and deal with are default vue events, which you can learn about [here](https://vuejs.org/v2/guide/events.html).
+
+However, for the sake of the two-way data-binding, we're sometimes firing events, which looks like this:
+```javascript
+this.$emit('some-event', additionalData);
+```
+
+Therefore you can also find those occurrences by searching for `$emit` in the `<shopware root>/platform/src/Storefront/Resources/app/storefront/src` directory for
```

Comment:
I think you have some :clipboard: :spaghetti: spilled over your docs. When you look for administration events you look into the administration code, right? :detective:

Reply:
I done goofed. 😱